### ISSUE TITLE
Layout/IndentationWidth not working as expected

### ISSUE THREAD

**Issue by Splines** [rubocop/rubocop#14946]
This relates to [`Layout/IndentationWidth`](https://docs.rubocop.org/rubocop/cops_layout.html#layoutindentationwidth), where `EnforcedStyleAlignWith` is set to `start_of_line` by default. However, this option does not seem to be respected and `relative_to_receiver` is enforced instead.

## 🎈 Example

We are using the following layout rules:

```yaml
# Extract of .rubocop.yml
Layout/EndAlignment:
  EnforcedStyleAlignWith: variable

Layout/IndentationConsistency:
  EnforcedStyle: indented_internal_methods

Layout/IndentationWidth:
  EnforcedStyleAlignWith: start_of_line

Layout/LineLength:
  Max: 100
```

```rb
class TestIndentation
  def abc
    Abc.where(id: "hello")
       .find_each do |user|
      @commontator_thread.subscribe(user)
    end
  end
end
```

We expected that RuboCop does not complain about this example.

## 🎈 Actual behavior

The given example is autocorrected to:

```rb
class TestIndentation
  def abc
    Abc.where(id: "hello")
       .find_each do |user|
         @commontator_thread.subscribe(user) # note the indentation of this line
    end
  end
end
```

## RuboCop version

```
1.84.2 (using Parser 3.3.10.2, Prism 1.9.0, rubocop-ast 1.49.0, analyzing as Ruby 3.4, running on ruby 3.4.7) [x86_64-linux]
  - rubocop-performance 1.26.1
  - rubocop-rails 2.34.3
```

**Comment by Splines** [comment]
I can reproduce this in versions `1.84.2` and `1.84.1`. With version `1.84.0` the formatting is as we expect, but you get this error in VSCode from the Ruby LSP extension:

```
Error running diagnostics: undefined method 'source_buffer' for an instance of RuboCop::Cop::Offense::PseudoSourceRange
```

This error is also present in `1.82.1`. In `1.82.0`, we get this instead:

```
Formatting error: An internal error occurred - Infinite loop detected in /workspaces/mampf/app/models/test-rubocop.rb and caused by Layout/IndentationWidth -> Layout/IndentationWidth. Updating to a newer version of RuboCop may solve this. For more details, run RuboCop on the command line.
```

`1.81.7` is the last version where the example works flawlessly. We have pinned that version.

By the way: why are there no `1.83.x` versions, i.e. a jump from `1.82.1` to `1.84.0`?

**Comment by koic** [comment]
It has not been released yet, but this issue has already been resolved on the latest `master` branch. Please wait for the next release.

The fact that RuboCop 1.83 was skipped was not intentional; it was likely due to an issue with the automation script used for the release process. Therefore, the next version after 1.82 will be 1.84.

**Comment by bbatsov** [comment]
> The fact that RuboCop 1.83 was skipped was not intentional; it was likely due to an issue with the automation script used for the release process. Therefore, the next version after 1.82 will be 1.84.

Indeed, it was once again an issue with the script - it bumped the version twice due to an error in the first run, and I noticed this too late. We'll need to figure out how to make this bulletproof eventually.
